src/main/java/uk/gov/hmcts/reform/sandl/snlapi/controllers/HearingController.java

```diff
@@ -35,4 +40,16 @@ public ResponseEntity assignHearingPartToSession(
         @RequestBody String assignment) {
         return eventsCommunicationService.makePutCall("/hearing/{hearingId}", assignment, hearingId);
     }
+
+    @PostMapping(path = "", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public String searchHearings(@RequestParam(value = "isListed", required = false) Optional<Boolean> isListed,
+                                  @RequestParam(value = "page", required = false) Optional<Integer> page,
+                                  @RequestParam(value = "size", required = false) Optional<Integer> size,
+                                    @RequestBody String searchCriteria) {
+        String url = "/hearing" +  (isListed.isPresent() ? "?isListed=" + isListed.get() : "");
+        url += (page.isPresent() && size.isPresent()) ? "?page=" + page.get() + "&size=" + size.get() : "";
```

Comment:
page is always concatenated with ?, I think you will lose isListed value if you add page.
Query string will look like ?isListed=?page=&size=

Reply:
done